chains/manager_integration_test.go

```diff
@@ -0,0 +1,56 @@
+// Copyright (C) 2019-2021, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+//go:build integration
+// +build integration
+
+package chains
+
+import (
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/flare-foundation/flare/ids"
+	"github.com/flare-foundation/flare/snow"
+	"github.com/flare-foundation/flare/snow/validation"
+	"github.com/flare-foundation/flare/utils/logging"
+	"github.com/flare-foundation/flare/vms/rpcchainvm"
+)
+
+func TestManagerIntegration(t *testing.T) {
+	cfg, _ := logging.DefaultConfig()
+	log, _ := logging.NewTestLog(cfg)
+	ctx := &snow.Context{
+		Log: log,
+	}
+
+	vmFactory := &rpcchainvm.Factory{
+		Path: filepath.Join("../build/plugins/mock"),
+	}
+	vm, err := vmFactory.New(ctx)
+	require.NoErrorf(t, err, "if this error occurs during testing, it is most likely that you need to run the `build_plugin_mock.sh` script")
+
+	m := &manager{
+		ManagerConfig: ManagerConfig{
+			Validators: validation.NewSet(),
+		},
+	}
+
+	r, ok := vm.(validation.Retriever)
+	require.True(t, ok)
+	ctx.ValidatorsRetriever = validation.NewCachingRetriever(r)
+	ctx.ValidatorsUpdater = validation.NewUpdater(m.Validators, ctx.ValidatorsRetriever)
+
+	got, err := ctx.ValidatorsRetriever.GetValidators(ids.ID{})
+	require.NoError(t, err)
+
+	assert.Equal(t, got.Len(), 9)
+	for i, v := range got.List() {
+		id := v.ID()
+		assert.Equal(t, byte(i)+1, id[0])
```

Comment:
I didn't get how it became `+1` as `i` always went from 0 onwards in both assignment and checking?

Reply:
TBH I'm not sure why the first value is skipped. I need to look into it, because the mock should return 10 values from 0 to 9, and what we get is from 1 to 9. This could be indicative of a bug somewhere actually 😬

Reply:
Yes, looks like it. 😬
I tried running the test in debugging mode but not much success in figuring out the core reason. All look normal until the FetchValidators() call: 
`res, err := vm.client.FetchValidators(context.Background(), &vmproto.FetchValidatorsRequest{`

Beyond this it isn't possible to check in debug mode.

Reply:
Yeah unfortunately since it all happens in the plugin, I've also hit the same wall 🤔 I'll look more into it when I'm up to date on PR reviews, if you want we can even take a look at it together today, since you have a better idea than I do of how this all works

Reply:
The weight cant be 0 https://github.com/flare-foundation/flare/blob/fb5aa99800792895ef1155c5d5db79e14abaca6c/snow/validation/set.go#L170
So maybe thats why it gets excluded